### ISSUE TITLE
[no-use-before-define] false positive when exported class used in type definition

### ISSUE THREAD

**Issue by zuzusik** [bradzacher/eslint-plugin-typescript#142]
```ts
type TArrayOfTest = Array<Test>;

export class Test {}
```
leads to 

```
  1:27  error  'Test' was used before it was defined  typescript/no-use-before-define
```

with 
```json
"typescript/no-use-before-define": [
  "error", 
  {"functions": false, "typedefs": false, "classes": false}
],
```

**Note:**  without `export` it passes fine:
```ts
type TArrayOfTest = Array<Test>;

class Test {}
```

**Comment by bradzacher** [comment]
tested both cases against v0.13.0, and they both report no-use-before-define.


Unless i'm mistaken, this is the correct behaviour?

**Comment by bradzacher** [comment]
tested both cases against v0.13.0, and they both report no-use-before-define.

**Comment by zuzusik** [comment]
@bradzacher actually in this case the correct behavior is **not** to report, as config has it 

```
"typescript/no-use-before-define": [
  "error", 
  {"functions": false, "typedefs": false, "classes": false}
]
```

so, classes **shouldn't** be reported, but they are when class is exported

**Comment by zuzusik** [comment]
@bradzacher sorry, I was replying to first version of your comment which I received to email, which had `Unless i'm mistaken, this is the correct behaviour?`

now I see that you have considered provided config when working on fix

so I think we are fine now 👍 thanks!
